api/src/utils/csv-utils/csv-header-configs.ts

```diff
@@ -0,0 +1,113 @@
+import { z } from 'zod';
+import { CSVError, CSVParams } from './csv-config-validation.interface';
+/**
+ * Utility function to strip the CSV parameters down to the error properties.
+ *
+ * @param {CSVParams} params - The cell parameters
+ * @returns {*} {Pick<CSVError, 'cell' | 'header' | 'rowIndex'>} - The error properties
+ */
+export const getCSVParamsError = (params: CSVParams): Pick<CSVError, 'cell' | 'header' | 'rowIndex'> => {
+  return {
+    cell: params.cell,
+    header: params.header,
+    rowIndex: params.rowIndex
+  };
+};
+
+/**
+ * Utility function to validate a CSV cell using a Zod schema.
+ *
+ * @param {CSVParams} params - The cell parameters
+ * @param {z.ZodSchema} schema - The Zod schema
+ * @returns {*} {CSVError[]} - The cell validation errors
+ */
+export const validateZodCell = (params: CSVParams, schema: z.ZodSchema): CSVError[] => {
+  const errors: CSVError[] = [];
+
+  const parsed = schema.safeParse(params.cell);
+
+  if (!parsed.success) {
+    parsed.error.errors.forEach((error) => {
+      errors.push({
+        error: error.message,
+        solution: 'Update the cell value to match the expected type',
+        ...getCSVParamsError(params)
+      });
+    });
+  }
+
+  return errors;
+};
+
+/**
+ * Get the TSN header cell validator.
+ *
+ * Rules:
+ *  1. The cell must be a number greater than or equal to 0
+ *  2. The cell must be a real ITIS TSN (from the provided set)
+ *
+ * @param {Set<number>} tsns Set of allowed ITIS TSNs
+ * @returns {*} {(params: CSVParams) => CSVError[]} The validate cell callback
+ */
+export const getTsnCellValidator = (tsns: Set<number>): ((params: CSVParams) => CSVError[]) => {
+  return (params: CSVParams) => {
+    const cellErrors = validateZodCell(params, z.number().min(0));
```

Comment:
Is this check required? 

We already have a list of allowed tsns, which have come from the biohub endpoint. So if the `Number(params.cell)` value below is in that list, then it would have to be valid

Reply:
I guess maybe it depends on the error messaging.

if the user enters "moose", do we tell them "must be a number" or just always say "not a valid tsn"

Reply:
Might be a question for Macgregor, what the error message should be and matching solution text.